### ISSUE TITLE
`mypy_extensions.TypedDict` errors because when creating a class, it is inferred as an instance

### ISSUE THREAD

**Issue by libre-man** [PyCQA/pylint#1927]
### Steps to reproduce
1. Create a new variable `Dict1 = mypy_extensions.TypedDict('Dict1', {'a' : int})`.
2. Create a class `class Dict2(Dict1, Total=True): pass`
3. Run pyling

### Current behavior
Pylint gives an error for the class creation as inheriting from a non class is invalid.

### Expected behavior
No error as this valid.

### pylint --version output
Python 3.6.4rc1

**Comment by PCManticore** [comment]
Thanks for the report. It seems that `mypy_extensions` does some metaprogramming that pylint and its inference mechanism can't quite understand.

**Comment by socketpair** [comment]
```python
class UserRecordType(TypedDict):
    id: str
    login: str
```

```
E0239: Inheriting 'TypedDict', which is not a class. (inherit-non-class)
```

Pylint 2.6.0-2
Python 3.9
